daemon/cluster/listen_addr.go

```diff
@@ -0,0 +1,163 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+func resolveListenAddr(specifiedAddr string, defaultListenAddr string) (string, error) {
+	specifiedHost, specifiedPort, err := net.SplitHostPort(specifiedAddr)
+	if err != nil {
+		return "", fmt.Errorf("could not parse listen address %s", specifiedAddr)
```

Comment:
@tonistiigi: If I'm reading this right, it means that the error path here should never be reached, correct? (Because `validateAndSanitizeInitRequest` adds the default port if there is no port).

Are you happy with the current approach, or should we change it to accept an empty `ListenAddr`, which would be handled the same way as `0.0.0.0`?